main.tf

```
@@ -133,10 +157,12 @@ resource "aws_launch_template" "default" {
     }
   }
 
-  instance_type = var.instance_types[0]
+  instance_type          = var.instance_types[0]
+  key_name               = var.ec2_ssh_key
+  vpc_security_group_ids = local.security_group_ids
```

In case we get the security group list wrong, I do not want to specify it if we do not have to.
```suggestion
  key_name               = local.ec2_ssh_key
  vpc_security_group_ids = local.ec2_ssh_key != null ?  local.security_group_ids : null
```